app/jobs/export_fhir_job.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# ExportFhirJob: prepare a FHIR export for a client application
+class ExportFhirJob < ApplicationJob
+  queue_as :exports
+  include Sidekiq::Status::Worker
+  sidekiq_options retry: 0
+
+  def perform(current_client_application, download, params)
+    patient_ids = Jurisdiction.find_by(id: current_client_application[:jurisdiction_id])&.all_patients_excluding_purged&.pluck(:id)
+    return if patient_ids.nil?
+
+    store transaction_time: DateTime.now.utc
+
+    patient_query = { id: patient_ids }
+    patient_query[:updated_at] = (params[:since]..) unless params[:since].nil?
+    add_file(download, Patient.where(patient_query), 'Patient.ndjson')
+
+    resource_query = { patient_id: patient_ids }
+    resource_query[:updated_at] = (params[:since]..) unless params[:since].nil?
+    add_file(download, Assessment.where(resource_query), 'QuestionnaireResponse.ndjson')
+    add_file(download, History.where(resource_query), 'Provenance.ndjson')
+    add_file(download, Laboratory.where(resource_query), 'Observation.ndjson')
+    add_file(download, Vaccine.where(resource_query), 'Immunization.ndjson')
+    add_file(download, CloseContact.where(resource_query), 'RelatedPerson.ndjson')
```

Comment:
Not trying to be too extra, but how would you feel about tracking progress % and incrementing after each `add_file` call, and using that value in the controller ExportStatus method? Probably would a good idea to top out at 99 and not set it to 100 till the last line in this method, just the user never sees in-progress and "100%". Looks like the `sidekiq-status` gem you added [provides](https://github.com/kenaniah/sidekiq-status#tracking-progress-and-storing-data) an interface for this.

The incremental values would not have to be scientific (super accurately reflect the actual exact % of time), but more just give users confidence that the job is not stuck and progress is actually getting made. Plus, thinking this could serve useful in debugging this in case any issues come up.

Reply:
Added these in https://github.com/SaraAlert/SaraAlert/pull/987/commits/5e43859283d46a982023b4d0a5965c6e314d7174.